lib/openstax/rescue_from/controller.rb

```diff
@@ -0,0 +1,40 @@
+module OpenStax
+  module RescueFrom
+    module Controller
+      def self.included(base)
+        base.extend ClassMethods
+      end
+
+      module ClassMethods
+        def use_openstax_exception_rescue
+          rescue_from Exception do |exception|
+            RescueFrom.perform_rescue(exception: exception, listener: self)
+          end
+        end
+      end
+
+      def before_openstax_exception_rescue(proxy)
+        @message = proxy.friendly_message
+        @code = proxy.status_code
+        @error_id = proxy.error_id
+      end
```

Comment:
I don't think these calls need to happen in a before hook, do they?  They are only used in the after hook.  If an outside user overrode these before or after hooks, it doesn't impact anything that goes on in between the hooks.  So I think these three lines can move to `after_openstax_exception_rescue` and `after_openstax_exception_rescue` can be renamed.  The current name kind of implies that it is always called at the end of rescuing, but it isn't -- it is only called when exceptions aren't reraised.  A name that matches that circumstance would be preferred -- something like `when_do_not_reraise` but better.